snapcraft/storeapi/__init__.py

```diff
@@ -89,6 +89,22 @@ def _get_name_from_snap_file(snap_path):
     return snap_yaml['name']
 
 
+def _macaroon_auth(conf):
+    """Format a macaroon and its associated discharge.
+
+    :return: A string suitable to use in an Authorization header.
+
+    """
+    root_macaroon_raw = conf.get('macaroon')
+    unbound_raw = conf.get('unbound_discharge')
+    root_macaroon = macaroons.Macaroon.deserialize(root_macaroon_raw)
+    unbound = macaroons.Macaroon.deserialize(unbound_raw)
+    bound = root_macaroon.prepare_for_request(unbound)
+    auth = 'Macaroon root={}, discharge={}'.format(root_macaroon_raw,
+                                                   bound.serialize())
```

Comment:
This should check that it doesn't break framing.  See the [Launchpad version of this](http://bazaar.launchpad.net/~launchpad-pqm/launchpad/devel/view/head:/lib/lp/snappy/model/snapstoreclient.py#L55).